### PR TITLE
[TF FE]: Support complex tensors for PadV2 operations (#22955)

### PR DESCRIPTION
**Pull request by awayzjj**
Closes https://github.com/openvinotoolkit/openvino/issues/22955
### Details:
 - separate input into real and imaginary parts and handle them separately by padding real or imag parts of padding value to them and concatenate them finally.

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_Pad.py [hunk 1]
```diff
@@ -75,9 +80,12 @@ def create_pad_complex_net(self, input_shape, pads_values):
...
-        dict(input_shape=[1, 5, 10, 3], pads_values=[[1, 1], [0, 0], [4, 0], [1, 1]]),
+        dict(input_shape=[1, 50], pads_values=[[0, 1], [2, 3]], const_value=None, pad_op='Pad'),
+        dict(input_shape=[2, 20, 10], pads_values=[[0, 1], [2, 3], [4, 0]], const_value=None, pad_op='Pad'),
+        dict(input_shape=[1, 5, 10, 3], pads_values=[[1, 1], [0, 0], [4, 0], [1, 1]], const_value=None, pad_op='Pad'),
+        dict(input_shape=[1, 50], pads_values=[[0, 1], [2, 3]], const_value=(1.0, 0.0), pad_op='PadV2'),
+        dict(input_shape=[2, 20, 10], pads_values=[[0, 1], [2, 3], [4, 0]], const_value=(0.0, 1.0), pad_op='PadV2'),
+        dict(input_shape=[1, 5, 10, 3], pads_values=[[1, 1], [0, 0], [4, 0], [1, 1]], const_value=(1.0, 2.0), pad_op='PadV2'),
     ]
 
     @pytest.mark.parametrize("params", test_data_basic)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Pad.py:91]: please revert these changes in test and just create separate test suite class `TestComplexPadV2`

File: src/frontends/tensorflow_common/src/op/pad.cpp [hunk 2]
```diff
@@ -82,10 +83,39 @@ OutputVector translate_pad_op(const NodeContext& node) {
...
+        auto minus_one = make_shared<v0::Constant>(element::i32, Shape{1}, -1);
+        auto x_real = make_shared<v8::Gather>(input, gather_index_real, minus_one)->output(0);
+        auto x_imag = make_shared<v8::Gather>(input, gather_index_imag, minus_one)->output(0);
+
+        auto constant_complex_type_mark = as_type_ptr<ComplexTypeMark>(constant_value.get_node_shared_ptr());
+        auto constant_input = constant_complex_type_mark->input_value(0);
+        auto constant = ov::util::get_constant_from_source(constant_input);
+        std::vector<float> constant_vector = constant->cast_vector<float>();
+        auto constant_value_real = create_same_type_const_scalar<float>(input, constant_vector[0]);
+        auto constant_value_imag = create_same_type_const_scalar<float>(input, constant_vector[1]);
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/pad.cpp:105]: Let us do not do this because `constant_input ` not neccesarily to be `Constant` node and it can be under `ShapeOf` sub-graph. Let us also compute real and imag parts using `Gather` as above. This way we do not need to re-create `Constant` node

**Review by rkazants — COMMENTED** [review]
You move the right direction, keep this way:)

File: src/frontends/tensorflow_common/src/op/pad.cpp (hunk 2 above)
Comment (awayzjj) [reply src/frontends/tensorflow_common/src/op/pad.cpp]: @rkazants Thank you for your comments! I have incorporated your suggestions. Can you please review this PR again? 

Also, I'm not clear on the sentence "it can be under ShapeOf sub-graph" in your comments. Could you please explain it?

File: tests/layer_tests/tensorflow_tests/test_tf_Pad.py [hunk 3]
```diff
@@ -87,3 +87,49 @@ def test_pad_complex(self, params, ie_device, precision, ir_version, temp_dir, u
...
+            real = tf.raw_ops.Real(input=pad)
+            imag = tf.raw_ops.Imag(input=pad)
+            tf.raw_ops.Pack(values=[real, imag], axis=-1)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    test_data_basic = [
+        dict(input_shape=[1, 50], pads_values=[[0, 1], [2, 3]], const_value=(1.0, 0.0)),
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Pad.py:124]: please set const_value to be complex values like, for example:

```
const_value = [1 + 2j, 1j]
```

File: tests/layer_tests/tensorflow_tests/test_tf_Pad.py [hunk 4]
```diff
@@ -87,3 +87,49 @@ def test_pad_complex(self, params, ie_device, precision, ir_version, temp_dir, u
...
+    def create_pad_complex_net(self, input_shape, pads_values, const_value):
+        tf.compat.v1.reset_default_graph()
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            param_real = tf.compat.v1.placeholder(np.float32, input_shape, 'param_real')
+            param_imag = tf.compat.v1.placeholder(np.float32, input_shape, 'param_imag')
+            complex = tf.raw_ops.Complex(real=param_real, imag=param_imag)
+            paddings = tf.constant(pads_values, dtype=tf.int32)
+            real_part, imag_part = const_value
+            constant_values = tf.complex(real_part, imag_part)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Pad.py:113]:
```suggestion
            constant_values = tf.constant(const_value, dtype=tf.complex64)
```

**Review by rkazants — APPROVED** [review]
Looks good. Please apply last comments in tests

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/tensorflow_tests/test_tf_Pad.py (hunk 4 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_Pad.py:113]: nit, let us skip this comment since CI is almost to the full green

File: tests/layer_tests/tensorflow_tests/test_tf_Pad.py (hunk 3 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_Pad.py:124]: nit, let us skip this comment since CI is almost to the full green

**Comment by rkazants** [comment]
build_jenkins

**Comment by rkazants** [comment]
@awayzjj, congratulations. PR was merged.
